pages/api/paybutton/download/transactions/[paybuttonId].ts

```diff
@@ -81,24 +88,32 @@ const formatNumberHeaders = (headers: string[], currency: string): string[] => {
 const downloadPaybuttonTransactionsFile = async (
   res: NextApiResponse,
   paybutton: PaybuttonWithAddresses,
-  currency: SupportedQuotesType): Promise<void> => {
-  const transactions = await fetchTransactionsByPaybuttonId(paybutton.id)
-
-  const mappedTransactionsData = transactions.sort((a, b) => {
-    return (a.timestamp - b.timestamp)
-  }).map(tx => {
+  currency: SupportedQuotesType,
+  networkTicker?: NetworkTickersType): Promise<void> => {
+  let networkIdArray = Object.values(NETWORK_IDS)
+  if (networkTicker !== undefined) {
+    const slug = Object.keys(NETWORK_TICKERS).find(key => NETWORK_TICKERS[key] === networkTicker)
+    const networkId = await getNetworkIdFromSlug(slug ?? NETWORK_TICKERS.ecash)
+    networkIdArray = [networkId]
+  }
+  const transactionsGrouped = await fetchTransactionsByPaybuttonIdGroupedByNetwork(paybutton.id, networkIdArray)
```

Comment:
Why do you need them grouped if you just ungroup it afterwards? This variable is not used anywhere further.

Reply:
this is to separate/group by network